tests/common/system_utils/docker.py

```diff
@@ -117,16 +117,13 @@ def swap_syncd(dut, registry_file=SONIC_DOCKER_REGISTRY):
             registry_file (str): The registry file describing where to download the RPC image.
     """
 
-    minigraph_facts = dut.minigraph_facts(host=dut.hostname)["ansible_facts"]
-    hw_sku = minigraph_facts["minigraph_hwsku"]
-
-    # TODO: Getting the vendor identifier should be a common utility
-    if is_broadcom_device(hw_sku):
+    asic_type = dut.facts["asic_type"]
+    if asic_type == "broadcom":
```

Comment:
I'd lean towards using the newly introduced API's here
if is_broadcom_device():
.
elif is_Mellanox_device():
.

Reply:
Ah, yes! I did it the long way because I was parsing the version every time. Let me fix that. :)

Reply:
Do you think it's OK to use the `dut.facts` in the log message or should we go back to having some sort of `get_asic_type` wrapper?

Reply:
This is tricky as it assumes we know how the is_<>_device API works. We can log **dut.facts as it has useful information about the device.